h/queue.py

```diff
@@ -26,18 +26,43 @@ def get_reader(settings, topic, channel):
     The caller is responsible for adding appropriate `on_message` hooks and
     starting the reader.
     """
-    addrs = aslist(settings.get('nsq.reader.addresses', 'localhost:4150'))
+    settings = request.registry.settings
+
+    sentry = getattr(request, 'sentry', None)
     topic = resolve_topic(topic, settings=settings)
+    addrs = aslist(settings.get('nsq.reader.addresses', 'localhost:4150'))
     reader = gnsq.Reader(topic, channel, nsqd_tcp_addresses=addrs)
+
+    if sentry is not None:
+        extra = {'topic': topic}
+
+        def _capture_exception(message, error):
+            if message is not None:
+                extra['message'] = message.body
+            sentry.captureException(exc_info=True, extra=extra)
+
+        def _capture_error(error):
+            sentry.extra_context(extra)
```

Comment:
Does this actually capture the error? From the Sentry docs it looks like it wouldn't.